dotnetv3/cross-service/ResilientService/AutoScalerActions/AutoScalerWrapper.cs

```diff
@@ -785,6 +905,36 @@ await _amazonAutoScaling.AttachLoadBalancerTargetGroupsAsync(
             });
     }
     // snippet-end:[ResilientService.dotnetv3.AutoScaling.AttachLoadBalancer]
+
+    /// <summary>
+    /// Wait until an EC2 instance is in a specified state.
+    /// </summary>
+    /// <param name="instanceId">The instance Id.</param>
+    /// <param name="stateName">The state to wait for.</param>
+    /// <returns>A Boolean value indicating the success of the action.</returns>
+    public async Task<bool> WaitForInstanceState(string instanceId, InstanceStateName stateName)
```

Comment:
Are there waiters for this? In Boto3 there seems to be a waiter for each instance state. Does .NET have the same?

Reply:
No, no waiters in .NET so we have to check the state ourselves.